src/core/render/index.js

```diff
@@ -409,10 +410,9 @@ export function Render(Base) {
         window.__current_docsify_compiler__ = this.compiler;
       }
 
-      const id = config.el || '#app';
-      const navEl = dom.find('nav') || dom.create('nav');
```

Comment:
We can solve this issue by using more specific CSS selectors instead of adding another configuration option:

* Desktop: `nav.app-nav`
* Mobile: `main > aside.sidebar > nav`

```js
 const navEl = dom.find('nav.app-nav, main > aside.sidebar > nav') || dom.create('nav');
```

Reply:
Won't this break a lot of existing configurations?

Reply:
Also, should this be a CSS id rather than class since there is probably only one `app-nav`?

Reply:
@jhildenbiddle that doesn't cover the use case that a person embeds docsify into a wrapper that has a nav on the outside, and that nav matching neither of those two selectors. That's what the main use case is here.

Reply:
@trusktr --

> that doesn't cover the use case that a person embeds docsify into a wrapper that has a nav on the outside, and that nav matching neither of those two selectors. That's what the main use case is here.

Perhaps we're interpreting the issue differently. Here's what @Evidlo said in #1525:

> I'm having a problem where docsify is hijacking my template's `<nav>` tag.

This is what the new selectors address: they prevent docsify from targeting the first `<nav>` element in the DOM by increasing the specificity. Based on the [demo](https://courses.engr.illinois.edu/ece445/wiki/#/) link provided, I'm assuming the goal is to prevent docsify from overwriting the existing top navigation (i.e., "hijacking" the orange banner with link to Home, Guidelines, Lab, etc.)

@Evidlo -- Let me know if I misinterpreted here.